### PR TITLE
ci: only run jobs when respected code changes

### PR DESCRIPTION
**Pull request by garbas**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,10 +14,48 @@ concurrency:
...
+    outputs:
+      pkgdb: "${{ steps.changes.outputs.pkgdb }}"
+      env-builder: "${{ steps.changes.outputs.env-builder }}"
+      cli: "${{ steps.changes.outputs.cli }}"
+
+    steps:
+      - uses: dorny/paths-filter@v2
+        id: changes
+        with:
+          filters: |
```
Comment (garbas) [inline .github/workflows/ci.yml:30]: @aakropotkin I'm not 100% sure about the paths which should trigger the rebuilds I'm sure you will know better. I guess you also need to first finish the top-level Makefile work, but I went ahead and create this PR anyway.

**Comment by garbas** [comment]
I've allowed `dorny/paths-filter@*` in the repo settings.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (aakropotkin) [reply .github/workflows/ci.yml:30]: Yeah wait for the top level Makefile. I plan to PR it tomorrow.

It's basically "anything under the subsystem name redoes the subsystem and consumers".

There's some magic top level files that trigger everything.

There's some boring files at the top level that are cosmetic/functionally irrelevant.

**Comment by garbas** [comment]
waiting until top-level Makefile work is merged  we'll take a clue from there if we need to rebuild something

**Comment by tomberek** [comment]
Merged a minimal form of this today. Closing as we can iterate from main.

**Pull request closed without merging**